packages/opentelemetry-api/src/context/propagation/TextMapPropagator.ts

```diff
@@ -54,8 +52,77 @@ export interface TextMapPropagator {
    *     the wire.
    * @param carrier the carrier of propagation fields, such as http request
    *     headers.
-   * @param getter a function which accepts a carrier and a key, and returns
-   *     the value from the carrier identified by the key.
+   * @param getter an optional {@link Getter}. If undefined, keys will be all
+   *     own properties, and keys will be accessed by direct object access.
    */
-  extract(context: Context, carrier: unknown, getter: GetterFunction): Context;
+  extract(context: Context, carrier: Carrier, getter: Getter<Carrier>): Context;
+}
+
+/**
+ * A setter is specified by the caller to define a specific method
+ * to set key/value pairs on the carrier within a propagator.
+ */
+export interface Setter<Carrier = any> {
```

Comment:
should the interfaces lives in `types` file or maybe the Noop ?

Reply:
The interface was originally in its own file, but I think it makes sense to keep it more tightly coupled to the TextMapPropagator since this getter is specifically for this type of propagator. If other propagator types are added in the future like binary they may have different interfaces.

Reply:
I'm fine with keeping the interfaces together with class if they are not exported or keeping them in some general file like `types.ts`. Otherwise I would rather keep them in separate file or in general `types.ts` as it quickly gets messy later with any refactoring or when trying to find the desired interface. 
Because of that:
> If other propagator types are added in the future like binary they may have different interfaces.

Can you change names:
- Setter into TextMapSetter
- Getter into TextMapGetter
- defaultGetter into defaultTextMapGetter

and then mentioned `types.ts` should become `textMapTypes.ts` ?
WDYT ?

Reply:
The TextMapPropagator is already just an interface. Won't it be weird to have a `types` file, but then also have some types (the `TextMapPropagator` interface) defined in a different file? I agree with the renaming though.

Reply:
you right it is interface already I was blind :/

Reply:
I did the rename as I think that is a good idea to prevent future incompatibilities